app/src/test/java/tttTDD/Java/BoardTest.java

```diff
@@ -33,17 +35,32 @@ public void getBoardSpace() {
     }
 
     @Test
-    public void updateBoardSpaceWithNewValue() throws Exception {
+    public void checkIfLocationIsOutOfBounds() {
+        board = new Board(3);
+
+        Assert.assertNull(board.getSpace(9));
+        Assert.assertNull(board.getSpace(16));
+    }
+
+    @Test
+    public void updateBoardSpaceWithNewValue(){
         board = new Board();
-        board.updateBoard(2, "X");
 
+        Assert.assertTrue(board.updateBoard(2, "X"));
         Assert.assertEquals(board.getSpace(2), "X");
     }
 
-    @Test(expected = Exception.class)
-    public void updatingTheSameSpaceWillResultInAnErrorException() throws Exception {
+    @Test
+    public void updatingTheSameSpaceWillResultInAnErrorException() {
         board = new Board();
         board.updateBoard(3, "X");
         board.updateBoard(3, "O");
```

Comment:
This call is both unnecessary, and demonstrates the down side to getting rid of the exception. At first glance it appears as if you made a valid move here but in reality nothing happens.

Reply:
I don't disagree. But want to point out, that it is only flawed due to not consuming the response after switching the implementation to be response-based. I do not yet see it as an inherent flaw not using exceptions.